Source/sheetworkers.js

```
@@ -365,33 +376,127 @@ function setupDivineFavorQuery(attrs) {
+      const dieFormula = dieSources
+        .map(([name, die]) => `${die}[${name}]`)
+        .join(" + ");
+      const harmType = Array.from(new Set(harms)).sort();
+      const formula = `{{roll=[[{${dieFormula}}k1 + @{strife_level}[${getTranslation("strife_level")}]]]}} {{harm=${harmType}}}`;
+      character[`strife_formula`] = formula;
+    });
+  },{repeating: {characters: ["strife_formula", ...getStrifeDiceAttributes()]}});
+}
+
+function safeGuardDiceSize(roll) {
+  return roll.toLowerCase().replace(/\b[^2]?d/g, 'd');
+}
+function joinNameAndTags(characters) {
+  let joined = [];
+  characters.forEach(character => {
+    let joinedChar = {};
+    if(character.tags) {
+      joinedChar.rolls = JSON.parse(character.tags);
+      joinedChar.rolls.forEach(roll => {
+        if(roll.dice) roll.dice = safeGuardDiceSize(roll.dice);
+        if(roll.harm) roll.harm = Object.keys(roll.harm);
+      });
+    }
+    let firstTag = {};
+    firstTag.name = character.name;
+    firstTag.dice = safeGuardDiceSize(character.dice);
+    if(character.harm) firstTag.harm = Object.keys(JSON.parse(character.harm));
+    joinedChar.rolls = [firstTag, ...joinedChar.rolls];
+    if(character.notes) joinedChar.notes = character.notes;
+    joined.push(joinedChar);
+  });
+  return joined;
+}
+function handleIsland() {
+  console.log("Handling Island Drop.");
+  getSetAttrs(["drop_name", "drop_data", "drop_content", "character_id", "token_src", "token_size", "sheet_type"], (attrs, setter) => {
+    let pagedata = {};
+    try {
+      pagedata = JSON.parse(attrs.drop_data);
+    } catch(e) {
+      console.log("Invalid drop-Data attribute");
+      pagedata = attrs.drop_data;
+    }
+    let category = pagedata["Category"];
+    let characters = [];
+    try {
+      characters = joinNameAndTags(JSON.parse(pagedata["data-characters"]));
+    } catch(e) {
+      console.log("Invalid data-Character page attribute");
```

Same here, we should reraise instead of continuing.